sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs

```diff
@@ -224,12 +224,13 @@ public IPAddress GetIPAddress()
                 new Uri(config.ActiveDirectoryAuthEndpoint));
 
         public TokenCredential GetOAuthCredential(string tenantId, string appId, string secret, Uri authorityHost) =>
-            new ClientSecretCredential(
+            Mode == RecordedTestMode.Playback ?
+            new TestCredential() :
+            (TokenCredential) new ClientSecretCredential(
                 tenantId,
                 appId,
                 secret,
-                Recording.InstrumentClientOptions(
-                    new TokenCredentialOptions() { AuthorityHost = authorityHost }));
+                new TokenCredentialOptions() { AuthorityHost = authorityHost });
```

Comment:
Style nit - indent the conditions for a multi-line ternary.  And maybe put the `(TokenCredential)` cast on the simpler case?